### DISCUSSION TITLE
Events Color

### DISCUSSION THREAD

**Discussion by dthanh267** [MMRIZE/MMM-CalendarExt3#171] [Q&A]
How do I change the color for different events?

**Comment by eouia** [comment]
Sorry for not getting back to you sooner, I was on vacation.
This module needs an event provider, usually the default `calendar` module. 
You can assign the color for each calendar like this.
```js
/* config.js */
{
	module: "calendar",
	header: "My Calendar",
	position: "top_left",
	config: {
		calendars: [
			{
				symbol: "calendar-check",
				url: "https://ics.calendarlabs.com/76/mm3137/US_Holidays.ics",
				color: "red",
			},
			{
				url: "https://...",
				color: "orange",
				symbol: "medal"
			},
		]
	}
},
```
This allows the events from each calendar to have different colors.

If you want to assign a certain color to each specific event, you may need `eventTransformer` property in your CX3 config. But generally, former method would be enough.

**Reply by dthanh267** [reply]
No worries, I've been reading post #124 and got some of things figured out, but I'm still unable to assign the color on specific events. In this one, Gus' event works for me, but not Logan's.

/* config.js */
{
	module: "MMM-CalendarExt3",
	position: "bottom_bar",
	config: {
		mode: "month",
		displayLegend: true,
		eventTransformer: (event) => {
			if (event.title.includes('Gus')) event.class = "gus"
			return event
			},
		eventTransformer: (event) => {
			if (event.title.includes('Logan')) event.class = "logan"
			return event
			},
		}
},

/* custom.css */

.CX3 .event.gus {
  background-color: var(--calendarColor);
  color: var(--oppositeColor);
  border-radius: 4px;
}

.CX3 .event.singleday.gus .headline .title {
  color: var(--oppositeColor);
}

.CX3 .event.singleday.gus .headline.useSymbol .symbol {
  color: var(--oppositeColor);
}

.CX3 .event.logan {
  background-color: var(--calendarColor);
  color: var(--oppositeColor);
  border-radius: 4px;
}

.CX3 .event.singleday.logan .headline .title {
  color: var(--oppositeColor);
}

.CX3 .event.singleday.logan .headline.useSymbol .symbol {
  color: var(--oppositeColor);
}
